<launch>
    <arg name = "drone_name" default = "drone"/>
    <arg name = "takeoff_height" default = "1.1"/>

    <!-- RATE,ATTI,POSY  -->
    <arg name = "ctrl_pub_level" default = "ATTI"/>
    <!-- RATE,ATTI,POSY  -->
    <arg name = "ctrl_mode" default = "POSY"/>
    


    <node pkg="tf" type="static_transform_publisher" name="map_to_world" args="0 0 0 0 0 0 1 map world 100" />
    <group ns="drone1">

    <node name = "ctrl_bridge" pkg = "emnv_ctl_bridge" type = "ctrl_bridge" output = "screen" required="true">
        <remap from="~pos_sp_cmd" to="/traj_test/cmd"/>
        <!-- <remap from="~vrpn_pose" to="/vrpn_client_node/emnavi0000/pose"/> -->
        
        <remap from="~ref_odom" to="/mavros/local_position/odom"/>
        <!-- Warning: /mavros/local_position/odom 的vel 是机体坐标系的，需要坐标转换，请将enable_vel_transpose_b2w设置为true -->
        <param name ="enable_vel_transpose_b2w" value="true"/>
        <param name ="drone_config_path" value="$(find emnv_ctl_bridge)/config/drone_gazebo_iris.yaml"/>

        <param name ="drone_name" value="$(arg drone_name)"/>
        <param name ="takeoff_height" value="$(arg takeoff_height)"/>
        <param name ="ctrl_pub_level" value="$(arg ctrl_pub_level)"/>
        <param name ="ctrl_mode" value="$(arg ctrl_mode)"/>

    </node> 
    <!-- 不要使用GSC_URL -->
    <!-- v1.13.3 版本对应的 gazebo 有问题，无法起飞 -->
        <include file="$(find mavros)/launch/px4.launch" >
            <arg name="fcu_url" value="udp://:14540@127.0.0.1:1857"/>
            <arg name="log_output" value="log"/>
        </include>
    </group>
</launch>
